document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/data/accounts/AccountsManager.java

```diff
@@ -1,44 +1,70 @@
 import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.stereotype.Component;
 import org.springframework.web.client.RestTemplate;
 import uk.gov.companieshouse.api.model.accounts.Accounts;
 import uk.gov.companieshouse.environment.EnvironmentReader;
 import uk.gov.companieshouse.environment.impl.EnvironmentReaderImpl;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
 
 /**
  * Temporary solution until private-sdk has been completed (SFA-518, SFA-670). When completed, this
  * file will get removed alongside the data package and all references to this file will be replaced
  * with calls to the private-sdk.
  */
+@Component
 public class AccountsManager {
 
     /** represents the Authorization header name in the request */
     private static final String AUTHORIZATION_HEADER = "Authorization";
 
     private static final EnvironmentReader READER = new EnvironmentReaderImpl();
 
-    private static String API_URL = READER.getMandatoryString("API_URL");
-    private static String CHS_API_KEY = READER.getMandatoryString("CHS_API_KEY");
+    private final String API_URL = READER.getMandatoryString("API_URL");
+    private final String CHS_API_KEY = READER.getMandatoryString("CHS_API_KEY");
 
     private static final RestTemplate restTemplate = createRestTemplate();
 
+    private static final Logger LOG = LoggerFactory.getLogger(MODULE_NAME_SPACE);
+
     /**
      * Get accounts resource if exists
      *
      * @param link - self link for the accounts object
      * @return accounts object along with the status or not found status.
+     * @throws Exception - throws a generic exception to mimic the private sdk throwing an exception.
+     *                     We're not to create a custom exception as it will have to be removed when
+     *                     the private sdk  gets implemented - additionally the generic exception is
+     *                     sufficient
      */
-    public static ResponseEntity<Accounts> getAccounts(String link) {
+    public Accounts getAccounts(String link) throws Exception {
         HttpHeaders requestHeaders = new HttpHeaders();
         requestHeaders.set(AUTHORIZATION_HEADER, getApiKey());
 
         HttpEntity requestEntity = new HttpEntity(requestHeaders);
-        return restTemplate.exchange(getRootUri() + link, HttpMethod.GET, requestEntity, Accounts.class);
+
+        ResponseEntity<Accounts> accountsResponseEntity = restTemplate.exchange(getRootUri() + link, HttpMethod.GET, requestEntity, Accounts.class);
+
+        if (accountsResponseEntity.getStatusCode() != HttpStatus.OK) {
+            Map<String, Object> logMap = new HashMap<>();
+            logMap.put("resource", link);
+            logMap.put("data.status", accountsResponseEntity.getStatusCode());
```

Comment:
You only need to use the key "status" here. "data" will automatically get prefixed.